deployer/__main__.py

```diff
@@ -19,6 +18,31 @@
 yaml = YAML(typ="safe", pure=True)
 
 
+def setup_cluster_credentials(cluster_name):
+    """
+    Quickly gain command-line access to a cluster by updating the current
+    kubeconfig file to include the deployer's access credentials for the named
+    cluster and mark it as the cluster to work against by default.
+
+    This function is to be used with the `setup-cluster-credentials` CLI
+    command only - it is not used by the rest of the deployer codebase.
+    """
+
+    # Validate our config with JSON Schema first before continuing
+    validate(cluster_name)
+
+    config_file_path = (
+        Path(os.getcwd()) / "config/clusters" / f"{cluster_name}.cluster.yaml"
+    )
+    with open(config_file_path) as f:
+        cluster = Cluster(yaml.load(f))
+
+    # Cluster.auth() method has the context manager decorator so cannot call
+    # it like a normal function
+    with cluster.auth():
+        return None
+
```

Comment:
This is interesting but I am not sure how it is expected to work, @sgibson91. 
You call it from the CLI and then the kubeconfig is updated with the deployer credentials, right?
But when the context manager finishes it should remove the environment variables or restore the kubeconfig (ie. https://github.com/2i2c-org/infrastructure/blob/master/deployer/hub.py#L226-L232, except in this case: https://github.com/2i2c-org/infrastructure/blob/master/deployer/hub.py#L147-L162, where we are not getting the kubeconfig to the original state). Am I probably missing something here?